src/code/z_kankyo.c

```diff
@@ -1,94 +1,2499 @@
+
+        if (globalCtx->envCtx.unk_1F != waterLightsIndex) {
+            globalCtx->envCtx.unk_1F = waterLightsIndex;
+            globalCtx->envCtx.unk_20 = waterLightsIndex;
+        }
+    } else {
+        globalCtx->envCtx.blendIndoorLights = false; // instantly switch to water lights
+        globalCtx->envCtx.unk_BF = waterLightsIndex;
+    }
+}
+
+void Kankyo_DisableUnderwaterLights(GlobalContext* globalCtx) {
+    if (!globalCtx->envCtx.indoors) {
+        globalCtx->envCtx.unk_1F = D_8011FB34;
+        globalCtx->envCtx.unk_20 = D_8011FB34;
+    } else {
+        globalCtx->envCtx.blendIndoorLights = false; // instantly switch to previous lights
+        globalCtx->envCtx.unk_BF = 0xFF;
+        globalCtx->envCtx.unk_D8 = 1.0f;
+    }
+}
+
+void Kankyo_PrintDebugInfo(GlobalContext* globalCtx, Gfx** gfx) {
+    GfxPrint printer;
+    u32 time;
+    s32 pad;
+
+    GfxPrint_Init(&printer);
+    GfxPrint_Open(&printer, *gfx);
+
+    GfxPrint_SetPos(&printer, 22, 7);
+    GfxPrint_SetColor(&printer, 155, 155, 255, 64);
+    GfxPrint_Printf(&printer, "T%03d ", ((void)0, gSaveContext.totalDays));
+    GfxPrint_Printf(&printer, "E%03d", ((void)0, gSaveContext.bgsDayCount));
+
+    GfxPrint_SetColor(&printer, 255, 255, 55, 64);
+    GfxPrint_SetPos(&printer, 22, 8);
+    GfxPrint_Printf(&printer, "%s", "ZELDATIME ");
+
+    GfxPrint_SetColor(&printer, 255, 255, 255, 64);
+    time = gSaveContext.dayTime;
+    GfxPrint_Printf(&printer, "%02d", (u8)(45.0f / 2048.0f * time / 60.0f));
+
+    if ((gSaveContext.dayTime & 0x1F) >= 0x10 || gTimeIncrement >= 6) {
+        GfxPrint_Printf(&printer, "%s", ":");
+    } else {
+        GfxPrint_Printf(&printer, "%s", " ");
+    }
+
+    time = gSaveContext.dayTime;
+    GfxPrint_Printf(&printer, "%02d", (s16)(45.0f / 2048.0f * time) % 60);
+
+    GfxPrint_SetColor(&printer, 255, 255, 55, 64);
+    GfxPrint_SetPos(&printer, 22, 9);
+    GfxPrint_Printf(&printer, "%s", "VRBOXTIME ");
+
+    GfxPrint_SetColor(&printer, 255, 255, 255, 64);
+    time = ((void)0, gSaveContext.skyboxTime);
+    GfxPrint_Printf(&printer, "%02d", (u8)(45.0f / 2048.0f * time / 60.0f));
+
+    if ((((void)0, gSaveContext.skyboxTime) & 0x1F) >= 0x10 || gTimeIncrement >= 6) {
+        GfxPrint_Printf(&printer, "%s", ":");
+    } else {
+        GfxPrint_Printf(&printer, "%s", " ");
+    }
+
+    time = ((void)0, gSaveContext.skyboxTime);
+    GfxPrint_Printf(&printer, "%02d", (s16)(45.0f / 2048.0f * time) % 60);
+
+    GfxPrint_SetColor(&printer, 55, 255, 255, 64);
+    GfxPrint_SetPos(&printer, 22, 6);
+
+    if (gSaveContext.nightFlag) {
+        GfxPrint_Printf(&printer, "%s", "YORU"); // "night"
+    } else {
+        GfxPrint_Printf(&printer, "%s", "HIRU"); // "day"
+    }
+
+    *gfx = GfxPrint_Close(&printer);
+    GfxPrint_Destroy(&printer);
+}
+
+#define TIME_ENTRY (D_8011FB48[envCtx->unk_1F] + i)
+#define TIME_ENTRY1 (&D_8011FB48[envCtx->unk_1F][i])
+#define TIME_ENTRY2 (&D_8011FB48[envCtx->unk_20][i])
```

Comment:
these lines get pretty unruly `lightSettingsList[D_8011FB48[envCtx->unk_1F][i].unk_04].ambientColor[j]`
even when these time entry arrays are named it will probably need something. would be ok with removing the macros for now and agreeing on something later though

Reply:
i think also given that its a nonmatching, it will make tweaking the accesses easier for whoever takes this function on. 
its pretty weird that they are accessed in two different ways, but this was added by petrie so it apparently helped the match in some way